security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -787,7 +738,58 @@ private Header extractHeaderSetCookie(HttpResponse proxiedResponse) {
 
     @VisibleForTesting
     protected HttpResponse executeHttpRequest(CloseableHttpAsyncClient httpclient, HttpRequestBase proxyingRequest) throws IOException, TimeoutException, ExecutionException, InterruptedException {
-        Future<HttpResponse> future = httpclient.execute(proxyingRequest, null);
+        CompletableFuture<HttpResponse> future = new CompletableFuture<HttpResponse>();
+
+        PipedOutputStream pos = new PipedOutputStream();
```

Comment:
Mape `pos`, `pis`, and `channel` members of the `consumer` inner class? Also add `@Override` annotations to `onEntityEnclosed`, `onContentReceived`, and `releaseResources`?

Reply:
thank you !